### PR TITLE
[MOB-2912] - Firebase Crash Fix

### PR DESCRIPTION
**Pull request by Ayyanchira**
## 🔹 Jira Ticket(s) if any

https://iterable.atlassian.net/browse/MOB-2912

## ✏️ Description

1. Upgrading firebase to 20.3.0 to use FirebaseMessaging instead of FirebaseInstanceId to get Token.
2. Updated related methods.

### CODE REVIEW

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableFirebaseMessagingService.java [hunk 1]
```diff
@@ -86,7 +88,16 @@ public static boolean handleMessageReceived(@NonNull Context context, @NonNull R
      * Call this from a custom {@link FirebaseMessagingService} to register the new token with Iterable
      */
     public static void handleTokenRefresh() {
-        String registrationToken = FirebaseInstanceId.getInstance().getToken();
+        String registrationToken;
+        try {
+            registrationToken = Tasks.await(FirebaseMessaging.getInstance().getToken());
```
Comment (vbabenkoru) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableFirebaseMessagingService.java:93]: This code is duplicated. It's better to move it to a class accessible from both places where it's used and call that instead.

**Review by roninopf — APPROVED** [review]

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableFirebaseMessagingService.java [hunk 2]
```diff
@@ -86,11 +88,25 @@ public static boolean handleMessageReceived(@NonNull Context context, @NonNull R
...
+        String registrationToken = getFirebaseToken();
         IterableLogger.d(TAG, "New Firebase Token generated: " + registrationToken);
         IterableApi.getInstance().registerForPush();
     }
 
+    public static String getFirebaseToken() {
+        String registrationToken = null;
+        try {
+            registrationToken = Tasks.await(FirebaseMessaging.getInstance().getToken());
+        } catch (ExecutionException e) {
```
Comment (davidtruong) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableFirebaseMessagingService.java:100]: Is there a reason why we are only catching these 2 specific exceptions? What happens if other exceptions occur?

**Comment by davidtruong** [comment]
Other than the comment about the error handling, overall it looks good.

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableFirebaseMessagingService.java (hunk 2 above)
Comment (Ayyanchira) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableFirebaseMessagingService.java:100]: Will add a general exception handling as well. 👍
Tasks.await throws those two exception. Hence added these two only initially

**Review by davidtruong — APPROVED** [review]
LFTM

**Pull request merged**